suites/portal/GWAS++UITest.js

```diff
@@ -0,0 +1,224 @@
+
+    gwas.do.selectAncestry();
+    gwas.do.clickNextButton();
+    I.wait(3);
+
+    I.seeElement(gwas.props.SubmitDialogBox);
+    const jobName = gwas.do.enterJobName();
+    await gwas.do.submitJob();
+    gwas.do.checkJobStatus();
+});
+
+Scenario('Submit workflow Dichotomous Outcome - Continuous Variate Phenotype @GWASUI', async ({
+    I, home, users, gwas
+}) => {
+    I.useWebDriverTo('set window size', async ({ browser }) => {
+        browser.setWindowSize(1920, 1080);
+    });
+    home.do.goToHomepage();
+    await home.complete.login(users.mainAcct);
+    I.saveScreenshot('homepage.png');
+
+    gwas.do.goToGWASPage();
+    gwas.do.selectCohort();
+    gwas.do.attritionTable();
+    gwas.do.clickNextButton(); 
+
+    gwas.do.selectDichotomouosPhenotype();
+    I.wait(3);
+    gwas.do.selectDichotomouosPhenotypeValues();
+    I.seeElement(gwas.props.RenderedEulerDiagram);
+    gwas.do.enterPhenotypeName();
+    I.click(gwas.props.PhenotypeSubmitButton);
+    //check the attrition table for update
+
+    gwas.do.selectContinuousVariate();
+    I.wait(3);
+    gwas.do.selectFirstConcept();
+    I.click(gwas.props.AddButton);
+    gwas.do.selectContinuousVariate();
+    gwas.do.selectSecondConcept();
+    I.click(gwas.props.AddButton);
+    I.saveScreenshot('Dichotomous-ContinuousVariate.png');
+    gwas.do.clickNextButton();
+
+    gwas.do.selectAncestry();
+    gwas.do.clickNextButton();
+    I.wait(3);
+
+    I.seeElement(gwas.props.SubmitDialogBox);
+    const jobName = gwas.do.enterJobName();
+    await gwas.do.submitJob();
+    gwas.do.checkJobStatus();
+});
+
+Scenario('Submit workflow Dichotomous Outcome - Dichotomous Variate Phenotype @GWASUI', async ({
+    I, home, users, gwas
+}) => {
+    I.useWebDriverTo('set window size', async ({ browser }) => {
+        browser.setWindowSize(1920, 1080);
+    });
+    home.do.goToHomepage();
+    await home.complete.login(users.mainAcct);
+    I.saveScreenshot('homepage.png');
+
+    gwas.do.goToGWASPage();
+    gwas.do.selectCohort();
+    gwas.do.attritionTable();
+    gwas.do.clickNextButton(); 
+
+    gwas.do.selectDichotomouosPhenotype();
+    I.wait(3);
+    gwas.do.selectDichotomouosPhenotypeValues();
+    I.seeElement(gwas.props.RenderedEulerDiagram);
+    gwas.do.enterPhenotypeName();
+    I.click(gwas.props.PhenotypeSubmitButton);
+    //check the attrition table for update
+
+    gwas.do.selectDichotomouosVariate();
+    gwas.do.selectFirstValue();
+    gwas.do.selectSecondValue();
+    I.wait(5);
+    I.seeElement(gwas.props.RenderedEulerDiagram);
+    gwas.do.enterPhenotypeName();
+    I.click(gwas.props.AddButton);
+    I.saveScreenshot('Dichotomous-DichotomousVariate.png');
+    gwas.do.clickNextButton();
+
+    gwas.do.selectAncestry();
+    gwas.do.clickNextButton();
+    I.wait(3);
+
+    I.seeElement(gwas.props.SubmitDialogBox);
+    const jobName = gwas.do.enterJobName();
+    await gwas.do.submitJob();
+    gwas.do.checkJobStatus();
+});
```

Comment:
the tests up to this point share a lot of the same steps. Please consider making reusable blocks.

Reply:
E.g., this part repeats:
```js
    I.useWebDriverTo('set window size', async ({ browser }) => {
        browser.setWindowSize(1920, 1080);
    });
    home.do.goToHomepage();
    await home.complete.login(users.mainAcct);
    I.saveScreenshot('homepage.png');

    gwas.do.goToGWASPage();
    gwas.do.selectCohort();
    gwas.do.attritionTable();
    gwas.do.clickNextButton(); 
```
and then this part at the end:
```js
    gwas.do.selectAncestry();
    gwas.do.clickNextButton();
    I.wait(3);

    I.seeElement(gwas.props.SubmitDialogBox);
    const jobName = gwas.do.enterJobName();
    await gwas.do.submitJob();
    gwas.do.checkJobStatus();
```

Reply:
I have reused the same piece of code (tasks) as I want to treat each scenario as an individual scenario. I can create reusable blocks of code, but it would end up being a spaghetti code which gets really difficult to debug if there are failures. To avoid this complexity, I have separated it out and have repeated steps in each scenario.

Reply:
Ok, your call. It is still manageable.